### ISSUE TITLE
[New Concept Docs]: Unpacking & Multiple Assignment in Python

### ISSUE THREAD

**Issue by BethanyG** [exercism/python#3111]
This issue describes how to implement the `unpacking` and `multiple assignment`  **concept docs**.
You can find the related **concept exercise** issue [here](https://github.com/exercism/python/issues/3112)

If you have not yet contributed to concept documents, this issue will require some upfront reading to give you the needed background knowledge.

## ✅   Getting started

**`Please please read the docs before starting.`** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. 

**General Contributing Docs:**

- [Contributing to Exercism](https://exercism.org/docs/building) | [Exercism and GitHub](https://exercism.org/docs/building/github) | - [Contributor Pull Request Guide](https://exercism.org/docs/building/github/contributors-pull-request-guide)
- [What are those Weird Task Tags about?](https://exercism.org/docs/building/product/tasks)
- [Exercism Formatting and Style Guide](https://exercism.org/docs/building/markdown/style-guide)
- [Exercism Markdown Specification](https://exercism.org/docs/building/markdown/markdown)
- [Reputation](https://exercism.org/docs/using/product/reputation)  

**Documents on Language Tracks and Concepts:**

- [Building Language Tracks: An Overview](https://exercism.org/docs/building/tracks) 
- [What are Concepts?](https://exercism.org/docs/building/tracks/concepts)
- [Concept Specifications](https://exercism.org/docs/building/tracks/concepts) 

<br>

## 🎯   Goal

These concept docs are meant to teach an understanding/use of `unpacking` and the `*` (splat) and `**` (double splat) operators  in Python. 

<br>

## 💡   Learning objectives

- Understand/use `unpacking` through the use of `*` and `**` _prefix_ operators in various scenarios
  - `*` and `**` as _prefixes_ ..... not to be confused with `*` (_multiply_) and `**` (_exponentiation_) as _infix_, or mathematical operators
  - what happens in the process of "unpacking" - form, ordering, & iteration
  - use in arguments to `functions`
  - use in argument _capture_ for `functions` (_aka passing an arbitrary number of arguments -- *args * & **kwargs_)
  - use in defining `keyword only arguments`
  - use in iterable (_mainly `tuple` and `list`_) unpacking & packing
  - use in `dict` unpacking & packing
- Understand/use `unpacking` via `multiple assignment`
  - using `multiple assignment ` in place of `indexing`
  - using `multiple assigment` + `*`  in place of `slicing`
  - using "list-like" syntax & "tuple-like" syntax
  - unpacking plus "leftovers" via `*`
- Differences between straight  `multiple assignment` and `*` & `**`
- Deep unpacking

<br>

## 🚫  Out of scope

<details>
<summary><b>Concepts & Subjects that are Out of Scope</b> (<em>click to open</em>)</summary>
<br>

- `classes`
- `comprehensions`
- `comprehensions` in `lambdas`
- `map()`, `filter()` or `functools.reduce()` in a `comprehension`
-  `function-arguments` beyond explaining briefly how `*`, `**` work in function arguments, and how `*` works in requiring arguments.
- `functools` beyond `functools.reduce()`(_this will get its own exercise_)
- `generators`
- using an `assignment expression` or "walrus" operator (`:=`) alone or in a `lambda`

</details>

<br>

## 🤔  Concepts

<details>
<summary><b>Concepts & Related Concepts Covered</b> (<em>click to open</em>)</summary>
<br>

- `unpacking`
- `unpacking generalizations`
- `multiple assignment`

</details>

<br>

## ↩️   Prerequisites

These are the concepts/concept exercises the student should be familiar with before taking on/learning this concept.  

<details>
<summary>Prereqs  (<em>click to open</em>)</summary>
<br>

- `basics`
- `bools`
- `comparisons`
- `dicts`
- `dict-methods`
- `functions`
- `function-arguments`
- `higher-order-functions`
- `functional tools`
- Identity methods `is` and `is not`
- `iteration`
- `lists`
- `list-methods`
- `numbers`
- `sequences`
- `sets`
- `strings`
- `string-methods`
- `tuples`

</details>

<br>  

## 📚  Resources to refer to

<details>
<summary><b>Resources</b>  (<em>click to open</em>)</summary>
<br>

- [Trey Hunner: Asterisks in Python - what they are and how to use them](https://treyhunner.com/2018/10/asterisks-in-python-what-they-are-and-how-to-use-them/)
- [Trey Hunner: Multiple assignment and tuple unpacking improve Python code readability](https://treyhunner.com/2018/03/tuple-unpacking-improves-python-code-readability/)
- [Dan Bader: Python Nested Unpacking](https://dbader.org/blog/python-nested-unpacking)
- [PEP 3132:  Extended Iterable Unpacking](https://www.python.org/dev/peps/pep-3132/)
- [PEP 0448: Additional Unpacking Generalizations](https://www.python.org/dev/peps/pep-0448/)
- [Python Tutorial:  Arbitrary Argument Lists](https://docs.python.org/3/tutorial/controlflow.html#arbitrary-argument-lists) & [Python Tutorial: Unpacking Argument Lists](https://docs.python.org/3/tutorial/controlflow.html#unpacking-argument-lists)
- [Python Docs: Expression Lists](https://docs.python.org/3/reference/expressions.html#expression-lists)
- [Python Docs: Defining Functions](https://docs.python.org/3/tutorial/controlflow.html#defining-functions)
- [built-ins: Python Docs](https://docs.python.org/3/library/functions.html)

</details>
<br>

##  📁  Files to Be Created

<details open>
<summary><b>File Detail for this Exercise</b> (<em>click to collapse</em>)</summary>
<table>
<td>
<br>

Please see the following for more details on these files:  [concepts](https://exercism.org/docs/building/tracks/concepts)

* ### `links.json`

  For more information, see [concept links file](https://github.com/exercism/docs/blob/main/building/tracks/concepts.md#file-linksjson)

  -  The same resources listed in this issue can be used as a starting point for the [ `concepts/links.json`](https://github.com/exercism/docs/blob/main/building/tracks/concepts.md#file-linksjson)  file, if it doesn't already exist.
  -  If there are particularly good/interesting information sources for this concept that extend or supplement the concept exercise material & the resources already listed -- please add them to the `links.json` document.

* ### Concept `about.md`

  For more information, see [Concept `about.md`](https://github.com/exercism/docs/blob/main/anatomy/tracks/concepts.md#file-aboutmd)

  - This file provides information about this concept for a student who has completed the corresponding concept exercise.  It is intended as a reference for continued learning.  

* ### Concept `introduction.md`

  For more information, see [Concept `introduction.md`](https://github.com/exercism/docs/blob/main/building/tracks/concepts.md#file-aboutmd)

  * This can also be a summary/paraphrase of the  `about.md` document listed above, and will provide a brief introduction of the concept for a student who has **not yet** completed the associated concept or practice exercises.   It should contain a good summation of the concept, but not go into lots of detail.

* ### Concept `.meta/config.json` Entries
 
  For more information, see [Concept `.meta/config.json`](https://github.com/exercism/docs/blob/main/building/tracks/concepts.md#file-metaconfigjson)

  * This file is likely already stubbed out.  Remember to add a concept **blurb** of less than 350 characters.  Please also add your GitHub username to the "authors" array, and any contributor GitHub usernames to the "contributors" array.

<br>
</td>
</table>
</details>

## 🎶  Implementation Notes

- Example code  should **only use syntax & concepts introduced within these docs or one of the prerequisite concept exercises or documents.** Where possible, please use REPL formatting, unless you are demonstrating pseudo code or a long code block.
  Please **do not use** syntax not previously covered in prerequisite topics or exercises.  Please also follow [PEP8](https://www.python.org/dev/peps/pep-0008/) guidelines.
- Our markdown and JSON files are checked against [prettier](https://prettier.io/) .  We recommend [setting prettier up locally](https://prettier.io/docs/en/install.html) and running it prior to submitting your PR  to avoid any CI errors.

<br>

##  🆘  Next Steps & Getting Help

1.  **`If you'd like to work on this issue, comment saying "I'd like to work on this"`** (_there is no real need to wait for a response, just go ahead, we'll assign you and put a `[claimed]` label on the issue_).
2. If you have any questions while implementing, please post the questions as comments in here, or contact one of the maintainers on our Slack channel.

**Comment by BethanyG** [comment]
Assigning to myself, as I can't seem to assign directly to @meatball133 yet without them commenting.  @meatball133 -- just comment here, and then I can assign.  Or shout **NO!** if you don't want it.  😉
